product_docs/docs/tde/15/enabling_tde_pg_upgrade.mdx

```diff
@@ -0,0 +1,76 @@
+
+Here are some example use cases:
+
+| Source system                               | Target system                                                      |
+|---------------------------------------------|--------------------------------------------------------------------|
+| Unencrypted EDB Postgres Extended Server 16 | Encrypted EDB Postgres Extended Server 16                          |
+| Unencrypted PostgreSQL 15                   | Encrypted EDB Postgres Extended Server 16                          |
+| Unencrypted PostgreSQL 16                   | Encrypted EDB Postgres Extended Server 16                          |
+| Encrypted EDB Postgres Extended Server 15   | Encrypted EDB Postgres Extended Server 16 with new encryption keys |
+
+## Overview 
+
+1. Perform a backup of your system. 
+1. Install the target system version, for example, EDB Postgres Extended Sever, or EDB Postgres Advanced Server. 
+1. Initialize a target instance and enable TDE encryption. 
+1. Use `pg_upgrade` with the `--copy-by-block` option to upgrade to a TDE system.
+
+## Example 
+
+This section displays using `pg_upgrade` to transition from an unencrypted PostgreSQL 16 system to an encrypted 
+EDB PostgreSQL Extender Server 16 on a Linux platform.
+
+### Step 1 - Perform a backup of your source system 
+
+Perform a backup of your system. You can use [Barman](https://www.enterprisedb.com/docs/supported-open-source/barman/), 
+[pg_dumpall](https://www.postgresql.org/docs/current/app-pg-dumpall.html) or a different backup utility.
+
+### Step 2 - Install the required target system
+
+Install the product where you want to create the target system, in this example, 
+[EDB Postgres Extended Server on Linux](/pge/latest/installing/).
+
+### Step 3 - Initialize the target instance
+
+Initialize a cluster in the target instance. The cluster must:
+
+  * Use wrap and unwrap data encryption keys.
+  * Support data encryption with the `--data-encryption` option. 
+  * Be accessible through a different port than the source instance.
+
+See [Using initdb TDE options](/tde/latest/enabling_tde/#using-initdb-tde-options) and this 
+[example](/tde/latest/enabling_tde/#example) for more information.
+
+[not sure if we need to start a database server as well.]: #
+
+### Step 4 - Use `pg_upgrade` to enable TDE
+
+Use the `pg_upgrade` utility. Specify the source and target systems, and use the `--copy-by-block` option.
+
+1. Stop both servers:
+
+   ```shell
+   /usr/lib/postgresql/16/bin/pg_ctl -D /source_instance/confdir/ stop
+   /usr/lib/edb-pge/16/bin/pg_ctl -D /target_instance/confdir/ stop
+   ```
+
+1. Perfom `pg_upgrade`. Specify the source and target bin directories, the source and target configuration directories, 
+and the `-–copy-by-block` option. 
+
+   ```shell 
+   /usr/lib/edb-pge/16/bin/pg_upgrade -b /usr/lib/postgresql/16/bin -B /usr/lib/edb-pge/16/bin -d /source_instance/confdir/ -D /target_instance/confdir/ --copy-by-block
+   ```
+
```

Comment:
This is as far as I got with testing, when running this command I got: 
```
Performing Consistency Checks
-----------------------------
Checking cluster versions                                     ok
The source cluster lacks some required control information:
  data encryption version
  data encryption key length
  ```

Reply:
Tushar and Mahendra helped test this. It turns out it is a bug. Fix probably will be merged with the next minor version in May. 
https://enterprisedb.atlassian.net/browse/DB-2385